src/services/deduplicationService.ts

```diff
@@ -0,0 +1,133 @@
+import { EnteFile } from 'types/file';
+import { getEndpoint } from 'utils/common/apiUtil';
+import { getToken } from 'utils/common/key';
+import { logError } from 'utils/sentry';
+import { getLocalFiles } from './fileService';
+import HTTPService from './HTTPService';
+
+const ENDPOINT = getEndpoint();
+
+interface DuplicatesResponse {
+    duplicates: Array<{
+        fileIDs: number[];
+        size: number;
+    }>;
+}
+
+interface DuplicateFiles {
+    files: EnteFile[];
+    size: number;
+}
+
+class DeduplicationService {
+    public async getDuplicateFiles() {
+        try {
+            const dupes = await this.fetchDuplicateFileIDs();
+            let ids: number[] = [];
+            for (const dupe of dupes) {
+                ids = ids.concat(dupe.fileIDs);
+            }
+
+            const localFiles = await getLocalFiles();
+            const fileMap = new Map<number, EnteFile>();
+            for (const file of localFiles) {
+                fileMap.set(file.id, file);
+            }
+
+            const result: DuplicateFiles[] = [];
+
+            const missingFileIDs: number[] = [];
+
+            for (const dupe of dupes) {
+                const files: EnteFile[] = [];
+                for (const fileID of dupe.fileIDs) {
+                    if (fileMap.has(fileID)) {
+                        files.push(fileMap.get(fileID));
+                    } else {
+                        missingFileIDs.push(fileID);
+                    }
+                }
+
+                if (files.length > 1) {
+                    result.push({
+                        files,
+                        size: dupe.size,
+                    });
+                }
+            }
+
+            if (missingFileIDs.length > 0) {
+                logError(
+                    new Error(`Missing files: ${missingFileIDs}`),
```

Comment:
as we know these are files that are in the trash. so no point in logging them na? we can just ignore them?